kornia/filters/blur_pool.py

```diff
@@ -202,3 +202,47 @@ def _max_blur_pool_by_kernel2d(
     # blur and downsample
     padding: Tuple[int, int] = _compute_zero_padding((kernel.shape[-2], kernel.shape[-1]))
     return F.conv2d(input, kernel, padding=padding, stride=stride, groups=input.size(1))
+
+
+def edge_aware_blur_pool2d(
+    input: torch.Tensor,
+    kernel_size: int,
+    edge_threshold: float = 2.0,
+    edge_dilatation_kernel_size: int = 3,
+    epsilon: float = 1e-6,
+) -> torch.Tensor:
+    r"""Blur the input tensor while maintaining its edges.
+    Edge detection is done with the sobel filter, and blurring is done with a pool2d.
+
+    Args:
+        input: the input image to blur with shape :math:`(B, C, H, W)`.
+        kernel_size: the kernel size for max pooling.
+        edge_threshold: threshold for the edge decision rule; edge/non-edge.
+        edge_dilatation_kernel_size: the kernel size for dilating the edges.
+        epsilon: for numerical stability.
+
+    Returns:
+        The blurred tensor of shape :math:`(B, C, H, W)`.
+    """
+    input = F.pad(input, (2, 2, 2, 2), mode="reflect")  # pad to avoid artifacts near physical edges
+    blurred_input = blur_pool2d(input, kernel_size=kernel_size, stride=1)  # blurry version of the input
+
+    # calculate the edges (add epsilon to avoid taking the log of 0)
+    log_input, log_thresh = torch.log2(input + epsilon), torch.log2(torch.tensor(edge_threshold))
```

Comment:
don't we need epsilon in the threshold ?

Reply:
Basically it's up to user... but I'll emphasis in the doctoring that it should be positive.